src/nft/components/profile/list/utils.ts

```diff
@@ -123,69 +108,89 @@ export const getListings = (sellAssets: WalletAsset[]): [CollectionRow[], Listin
-    allCollectionsDefined = allCollectionsPending = false
-  }
-  for (const collection of collectionsRequiringApproval) {
-    if (collection.status !== ListingStatus.PENDING) allCollectionsPending = false
-    if (collection.status !== ListingStatus.DEFINED) allCollectionsDefined = false
-    if (collection.status === ListingStatus.SIGNING) anyActiveSigning = true
-    else if (collection.status === ListingStatus.FAILED) anyActiveFailures = true
-    else if (collection.status === ListingStatus.REJECTED) anyActiveRejections = true
-    else if (collection.status === ListingStatus.PAUSED) anyPaused = true
-  }
+export function useHandleGlobalPriceToggle(
+  globalOverride: boolean,
+  setListPrice: Dispatch<number | undefined>,
+  setPrice: (price?: number) => void,
+  listPrice?: number,
+  globalPrice?: number
+) {
+  useEffect(() => {
+    let price: number | undefined
+    if (globalOverride) {
+      if (!listPrice) setListPrice(globalPrice)
+      price = globalPrice
+    } else {
+      price = listPrice
+    }
+    setPrice(price)
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [globalOverride])
+}
 
-  if (listings.length === 0) {
-    allListingsApproved = allListingsDefined = allListingsPending = false
-  }
-  for (const listing of listings) {
-    if (listing.status !== ListingStatus.PENDING) allListingsPending = false
-    if (listing.status !== ListingStatus.DEFINED) allListingsDefined = false
-    if (listing.status !== ListingStatus.APPROVED) allListingsApproved = false
-    if (listing.status === ListingStatus.SIGNING) anyActiveSigning = true
-    else if (listing.status === ListingStatus.FAILED) anyActiveFailures = true
-    else if (listing.status === ListingStatus.REJECTED) anyActiveRejections = true
-    else if (listing.status === ListingStatus.PAUSED) anyPaused = true
-  }
-  return {
-    allListingsPending,
-    allListingsDefined,
-    allListingsApproved,
-    allCollectionsPending,
-    allCollectionsDefined,
-    anyActiveSigning,
-    anyActiveFailures,
-    anyActiveRejections,
-    anyPaused,
-  }
+export function useSyncPriceWithGlobalMethod(
+  asset: WalletAsset,
+  setListPrice: Dispatch<number | undefined>,
+  setGlobalPrice: Dispatch<number | undefined>,
+  setGlobalOverride: Dispatch<boolean>,
+  listPrice?: number,
+  globalPrice?: number,
+  globalPriceMethod?: SetPriceMethod
+) {
+  useEffect(() => {
+    if (globalPriceMethod === SetPriceMethod.FLOOR_PRICE) {
+      setListPrice(asset?.floorPrice)
+      setGlobalPrice(asset.floorPrice)
+    } else if (globalPriceMethod === SetPriceMethod.LAST_PRICE) {
+      setListPrice(asset.lastPrice)
+      setGlobalPrice(asset.lastPrice)
+    } else if (globalPriceMethod === SetPriceMethod.SAME_PRICE)
+      listPrice && !globalPrice ? setGlobalPrice(listPrice) : setListPrice(globalPrice)
+
+    setGlobalOverride(false)
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [globalPriceMethod])
```

Comment:
same question here do we not want this to update when listPrice changes?

Reply:
Yeah we want this atomic to only when the globalPriceMethod changes, otherwise it can fire prematurely and overwrite user input